spec/relayer/ics-018-relayer-algorithms/README.md

```diff
@@ -266,6 +266,723 @@ The relay process must have access to accounts on both chains with sufficient ba
+      "value": "{plan.height}"
+    }
+  ]
+}
+```
+
+#### Connection lifecycle
+
+Format of the event emitted when a connection end in `INIT` state is successfully added in 
+the `connOpenInit` handler:
+
+```json
+{
+  "type": "connection_open_init",
+  "attributes": [
+    {
+      "key": "client_id",
+      "value": "{clientId}"
+    },
+    {
+      "key": "connection_id",
+      "value": "{connectionId}"
+    },
+    {
+      "key": "counterparty_client_id",
+      "value": "{counterparty.clientId}"
+    },
+    {
+      "key": "counterparty_connection_id",
+      "value": "{counterparty.connectionId}"
+    }
+  ]
+}
+```
+
+Format of the event emitted when a connection end in `TRY` state is is successfully added in 
+the `connOpenTry` handler:
+
+```json
+{
+  "type": "connection_open_try",
+  "attributes": [
+    {
+      "key": "client_id",
+      "value": "{clientId}"
+    },
+    {
+      "key": "connection_id",
+      "value": "{connectionId}"
+    },
+    {
+      "key": "counterparty_client_id",
+      "value": "{counterparty.clientId}"
+    },
+    {
+      "key": "counterparty_connection_id",
+      "value": "{counterparty.connectionId}"
+    }
+  ]
+}
+```
+
+Format of the event emitted when an existing connection end in `INIT` is successfully updated 
+to `OPEN` state in the `connOpenAck` handler:
+
+```json
+{
+  "type": "connection_open_ack",
+  "attributes": [
+    {
+      "key": "client_id",
+      "value": "{clientId}"
+    },
+    {
+      "key": "connection_id",
+      "value": "{connectionId}"
+    },
+    {
+      "key": "counterparty_client_id",
+      "value": "{counterparty.clientId}"
+    },
+    {
+      "key": "counterparty_connection_id",
+      "value": "{counterparty.connectionId}"
+    }
+  ]
+}
+```
+
+Format of the event emitted when an existing connection end in `TRY` state is successfully updated 
+to `OPEN` state in the `connOpenConfirm` handler:
+
+```json
+{
+  "type": "connection_open_confirm",
+  "attributes": [
+    {
+      "key": "client_id",
+      "value": "{clientId}"
+    },
+    {
+      "key": "connection_id",
+      "value": "{connectionId}"
+    },
+    {
+      "key": "counterparty_client_id",
+      "value": "{counterparty.clientId}"
+    },
+    {
+      "key": "counterparty_connection_id",
+      "value": "{counterparty.connectionId}"
+    }
+  ]
+}
+```
+
+#### Channel lifecycle
+
+Format of the event emitted when a channel end in `INIT` state is successfully added in 
+the `handleChanOpenInit` datagram handler:
+
+```json
+{
+  "type": "channel_open_init",
+  "attributes": [
+    {
+      "key": "port_id",
+      "value": "{portId}"
+    },
+    {
+      "key": "channel_id",
+      "value": "{channelId}"
+    },
+    {
+      "key": "counterparty_port_id",
+      "value": "{counterparty.portId}"
+    },
+    {
+      "key": "counterparty_channel_id",
+      "value": "{counterparty.channelId}"
+    },
+    {
+      "key": "connection_id",
+      "value": "{connectionHops[0]}"
```

Comment:
Note this will need to change for multihop. Perhaps we can create a tracking issue